### PR TITLE
bridgeDeposits

### PR DESCRIPTION
**Pull request by ylv-io**
# Description

This PR refactors the bridging functionality within the Bridger contracts to use a new bridge interface and gateway.

- Modified the constructor of the `Bridger` contract to accept an additional `bridge` address parameter and initialize an immutable `bridgeGateway`.
- Updated the `bridgeDeposits` function in the `Bridger` contract to use the new `bridgeGateway` for bridging deposits instead of the previous `L1GatewayRouter`.
- Introduced a new `BridgeData` struct in the `IBridger` interface to encapsulate bridge-related parameters.
- Added a new `IBridge.sol` interface to define the functions of the bridge contract.

## Type of change

- [ ] Bug fix 
- [x] New feature 
- [x] Breaking change 
- [ ] Dependency changes
- [ ] Deployment
- [ ] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [x] The diff is legible and has no extraneous changes
- [x] Complex code has been commented, including external interfaces
- [ ] Tests have 100% code coverage
- [x] The base branch is either `main`, or there's a description of how to merge

### CODE REVIEW

File: src/bridger/Bridger.sol [hunk 1]
```diff
@@ -201,34 +203,24 @@ contract Bridger is
     }
 
     /**
-     * @dev Bridges deposits in bulk every hour to the L2
+     * @dev Bridges deposits to Kinto
```
Comment (rrecuero) [inline src/bridger/Bridger.sol:206]: depositETH and depositBySig should call bridge after the swap is made

File: src/bridger/Bridger.sol [hunk 2]
```diff
@@ -201,34 +203,24 @@ contract Bridger is
...
      */
-    function bridgeDeposits(address asset, uint256 maxGas, uint256 gasPriceBid, uint256 maxSubmissionCost)
-        external
-        payable
-        override
-        onlyPrivileged
-    {
+    function bridgeDeposits(address asset, BridgeData calldata bridgeData) external payable onlyPrivileged {
```
Comment (rrecuero) [inline src/bridger/Bridger.sol:208]: should bridgeData include amount so we can test first with 1 unit?

**Review by fedealconada — APPROVED** [review]
lgtm! thanks! :) looking forward to see this working

File: src/bridger/Bridger.sol (hunk 2 above)
Comment (ylv-io) [reply src/bridger/Bridger.sol:208]: Yep, lets do it, just to be extra safe!

**Pull request merged**